tests/register.spec.ts

```diff
@@ -181,6 +182,170 @@ test.describe.serial('should complete the registration process', () => {
+    await page.getByRole('button', { name: 'Edit Your Answer' }).first().click();
+
+    // Expect correct data for first question
+    await expect(page).toHaveURL(/(http[s]?:\/\/)?(.*)\/en\/candidate\/questions\/edit\/1/);
+    await expect(page.getByRole('heading', { name: 'The state should preferably' })).toBeVisible();
+    await expect(page.getByLabel('Fully agree')).toBeChecked();
+
+    // Check that button goes back to correct page
+    await page.getByRole('button', { name: 'Save and Return' }).click();
+    await expect(page).toHaveURL(/(http[s]?:\/\/)?(.*)\/en\/candidate\/questions/);
+
+    // Go back and test cancel button
+    await page.getByLabel('open ').nth(1).click();
+    await page.getByRole('button', { name: 'Edit Your Answer' }).first().click();
+
+    // Change opinion and press cancel
+    await page.getByLabel('Fully disagree').click();
+    await page.getByRole('button', { name: 'Cancel' }).click();
+
+    // Expect button to go to correct page and opinion to be unchanged
+    await expect(page).toHaveURL(/(http[s]?:\/\/)?(.*)\/en\/candidate\/questions/);
+    await page.getByLabel('open ').nth(1).click();
+    await expect(page.getByText('Fully agree').first()).toBeVisible();
+
+    // Expect return button to go to correct page
+    await page.getByRole('button', { name: 'Return' }).click();
+    await expect(page).toHaveURL(/(http[s]?:\/\/)?(.*)\/en\/candidate/);
+  });
+
+  test('preview page should work correctly', async ({ page }) => {
+
+    // Log the default user out
+    await page.goto(`http://localhost:5173/en/candidate`);
+    await page.getByLabel('Logout').click();
+
+    // Log in with the created user Alice
+    await page.getByLabel('Email').fill(userEmail);
+    await page.getByLabel(/^Password$/).fill(userPassword);
+    await page.getByText('Sign in').click();
+
+    // Go to preview page
+    await page.goto(`http://localhost:5173/en/candidate/preview`);
+    await page.reload(); //Reload to make sure correct data is loaded to page
+
+    // Expect correct title and button
+    await expect(page.getByText('This is a preview of how your')).toBeVisible();
+    await expect(page.getByRole('button', { name: 'Close preview' })).toBeEnabled();
+
+    // Expect correct data for the candidate 
+    await expect(page.getByRole('heading', { name: `${userFirstName} ${userLastName}` })).toBeVisible();
+    await expect(page.getByText('Mother Tongues Finnish • Spanish')).toBeVisible();
+    await expect(page.getByText(`Gender ${userGender}`)).toBeVisible();
+    await expect(page.getByText('Unaffiliated No')).toBeVisible();
+    await expect(page.getByText(`Election Manifesto ${userManifesto}`)).toBeVisible();
+    await expect(page.getByText('Birthday Mon Jan 01 1990')).toBeVisible();
+
+    // Expect close button to take to start page
+    await page.getByRole('button', { name: 'Close preview' }).click();
+    await expect(page).toHaveURL(/(http[s]?:\/\/)?(.*)\/en\/candidate/);
+  });
+
+});
```

Comment:
Make sure to add a new line to the end of the file. It should be automatically fixed, if you run prettier on this file